File: java-fullstack-app/domain/src/test/java/com/example/domain/model/OrderTest.java
package com.example.domain.model;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class OrderTest {

    @Test
    void newOrder_isPending() {
        Order order = new Order("1", 50.0);
        assertTrue(order.isPending());
        assertEquals("PENDING", order.getStatus());
    }

    @Test
    void setStatus_updatesStatus() {
        Order order = new Order("2", 100.0);
        order.setStatus("CONFIRMED");
        assertFalse(order.isPending());
        assertEquals("CONFIRMED", order.getStatus());
    }
}

File: node-mono-app/packages/core/src/__tests__/order.test.ts
import { isPending, Order } from '../order';

describe('isPending', () => {
  it('returns true for pending order', () => {
    const o: Order = { id: '1', total: 50, status: 'pending' };
    expect(isPending(o)).toBe(true);
  });

  it('returns false for confirmed order', () => {
    const o: Order = { id: '2', total: 50, status: 'confirmed' };
    expect(isPending(o)).toBe(false);
  });
});
